modules/billing-account/factory.tf

```diff
@@ -0,0 +1,98 @@
+/**
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+locals {
+  _factory_data = {
+    for f in fileset("${local._factory_path}", "**/*.yaml") :
+    trimsuffix(f, ".yaml") => yamldecode(file("${local._factory_path}/${f}"))
+  }
+  _factory_path = var.factory_config.budgets_data_path
+  factory_budgets = {
```

Comment:
it's quite difficult to read. I'd consider splitting this in multiple locals

Reply:
what is difficult to read Luca?